### PR TITLE
Contract Rework, Waw minor changes

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
You can now sign a contract with A Contract, Signed.
DRV damage increased by 50%.
Flesh idol healing doubled; It breaches one abno instead of two

## Why It's Good For The Game
Buff to what needs to be buffed, nerfs what need to be nerfed.

## Changelog
:cl:
add: You can now sign contracts with contracts
balance: nerfs Flesh Idol's breaching
balance: DRV damage increased by 50%
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm [hunk 1]
```diff
@@ -25,22 +25,55 @@
...
-
+	//We need a list of all abnormalities that are TETH to HE level and Can breach.
 	var/list/queue = subtypesof(/mob/living/simple_animal/hostile/abnormality)
 	for(var/i in queue)
 		var/mob/living/simple_animal/hostile/abnormality/abno = i
 		if(!(initial(abno.can_spawn)) || !(initial(abno.can_breach)))
 			continue
 
-		if((initial(abno.threat_level)) <= HE_LEVEL)
+		if((initial(abno.threat_level)) <= WAW_LEVEL)
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:46]: Im probably gonna have to retool SoW for this. Probably.

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm [hunk 2]
```diff
@@ -25,22 +25,55 @@
...
 			spawnables += abno
 
+/mob/living/simple_animal/hostile/abnormality/contract/WorkChance(mob/living/carbon/human/user, chance, work_type)
+	if(!(user in total_havers))
+		return chance
+
+	var/enabled
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:53]:
```suggestion
	var/enabled = FALSE
```
Literally just a code clarity thing. I know this works without this being set.

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm [hunk 3]
```diff
@@ -51,9 +84,48 @@
...
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+			if(ABNORMALITY_WORK_REPRESSION)
+				if(just_havers.len < total_per_contract)
+					user.adjust_attribute_buff(JUSTICE_ATTRIBUTE, (just_havers.len - 3)*-1 )
+					just_havers |= user
+					total_havers |= user
+					say("Just sign here on the dotted line... and I'll take care of the rest.")
+
+	return
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:122]:
```suggestion
	if((user in total_havers))
		return
	switch(work_type)
		if(ABNORMALITY_WORK_INSTINCT)
			if(fort_havers.len < total_per_contract)
				user.adjust_attribute_buff(FORTITUDE_ATTRIBUTE, (fort_havers.len - 3)*-1 )
				fort_havers |= user
			else
				return

		if(ABNORMALITY_WORK_INSIGHT)
			if(prud_havers.len < total_per_contract)
				user.adjust_attribute_buff(PRUDENCE_ATTRIBUTE, (prud_havers.len - 3)*-1 )
				prud_havers |= user
			else
				return

		if(ABNORMALITY_WORK_ATTACHMENT)
			if(temp_havers.len < total_per_contract)
				user.adjust_attribute_buff(TEMPERANCE_ATTRIBUTE, (temp_havers.len - 3)*-1 )
				temp_havers |= user
			else
				return
				
		if(ABNORMALITY_WORK_REPRESSION)
			if(just_havers.len < total_per_contract)
				user.adjust_attribute_buff(JUSTICE_ATTRIBUTE, (just_havers.len - 3)*-1 )
				just_havers |= user
			else
				return
				
	total_havers |= user
	say("Just sign here on the dotted line... and I'll take care of the rest.")
	return
```
This should reduce redundancy and also uses early returns to cut off unnecessary code and ideally improve readability.

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm [hunk 4]
```diff
@@ -25,22 +25,55 @@
 		)
 //	gift_type = /datum/ego_gifts/eight
 
+	var/list/total_havers = list()
 
+	var/list/fort_havers = list()
+	var/list/prud_havers = list()
+	var/list/temp_havers = list()
+	var/list/just_havers = list()
```
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:33]: Should these lists have their nulls and dead people cleaned out at any point?

**Review by dsmith328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm (hunk 4 above)
Comment (Kitsunemitsu) [reply code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:33]: It shouldn't, the idea is that you get 3 slight stat buffs, per stat, per round.

File: code/modules/mob/living/simple_animal/abnormality/waw/contract.dm (hunk 1 above)
Comment (Kitsunemitsu) [reply code/modules/mob/living/simple_animal/abnormality/waw/contract.dm:46]: Want me to wait?

**Review by dsmith328 — APPROVED** [review]

**Comment by Kitsunemitsu** [comment]
Which workrate update did we talk about?

**Pull request merged**